### PR TITLE
Ben/finetune cv

### PR DESCRIPTION
**Pull request by benleetownsend**
So the basic functionality of this is that the ranged hparams have their default value, but when run through grid search will use the range. I think this is neat because it means we can keep a central config of everything meaning nothing gets out of sync. This is slightly at the expense of readability. 

I still need to add actual cv splits instead of a static split. But this is just for a high level design overview.

Also needs docs.

TODOs

- [ ] Docs
- [ ] CV spits
- [ ] Tests

### CODE REVIEW

File: finetune/config.py [hunk 1]
```diff
@@ -19,6 +19,18 @@ def all_gpus():
     ]
 
 
+class Ranged:
+    def __init__(self, default, itterator=None):
+        pass
+
+    def __new__(self, default, itterator=None):
+        class Ranged_(type(default)):
+            def get_itterator(self):
```
Comment (madisonmay) [inline finetune/config.py]: mispelled

File: finetune/base.py [hunk 2]
```diff
@@ -715,3 +716,36 @@ def _load_finetuned_model(self):
         saver.restore(self.sess, os.path.join(self._load_from_file, SAVE_PREFIX))
         self._load_from_file = False
         self.is_trained = True
+
+    @classmethod
+    def finetune_grid_search(cls, config, Xs, Y, eval_fn, test_size, probs=False, return_all=False):
```
[the remark this answers is not in the document]
Comment (madisonmay) [reply finetune/base.py]: abc with default eval_fn

[the remark this answers is not in the document]
Comment (madisonmay) [reply finetune/base.py]: consider re-fit automatically after cv sweep

File: finetune/base.py [hunk 3]
```diff
@@ -715,3 +716,36 @@ def _load_finetuned_model(self):
...
+        trainXs, testXs, trainY, testY = train_test_split(list_transpose(Xs), Y, test_size=test_size)
```
Comment (madisonmay) [inline finetune/base.py]: could stochastically produce bad splits that raise errors

Comment (madisonmay) [reply finetune/base.py]: take a look at Enso here and mimic it's functionality

**Comment by benleetownsend** [comment]
Fixed PR stuff. UNTESTED

**Pull request merged**